architecture/additional_concepts/authentication.adoc

```diff
@@ -206,13 +206,136 @@ When using a service account as an OAuth client:
 $ oc sa get-token <serviceaccount name>
 ----
 
-* `redirect_uri` must match an annotation on the service account that has the
```

Comment:
Also up above out of this PR's view, a general comment: all the subsections in the "OAuth" section have been just bold headings for a long time. AFAICT I'm the one that did that long ago, but at any rate, they could probably be real subheadings now, so that they show up in TOC / are easier to link to. Just, while you're in this area already!